src/Jaeger/Sender/JaegerThriftSender.php

```diff
@@ -0,0 +1,294 @@
+                $span->getComponent() ?? $span->getTracer()->getServiceName(),
+                $endpoint
+            );
+
+            $span->tags[COMPONENT] = $tag;
+        }
+    }
+
+    private function makeLocalComponentTag(string $componentName, Endpoint $endpoint): BinaryAnnotation
+    {
+        return new BinaryAnnotation([
+            'key' => "lc",
+            'value' => $componentName,
+            'annotation_type' => AnnotationType::STRING,
+            'host' => $endpoint,
+        ]);
+    }
+
+    private function makeEndpoint(string $ipv4, int $port, string $serviceName): Endpoint
+    {
+        $ipv4 = $this->ipv4ToInt($ipv4);
+
+        return new Endpoint([
+            'ipv4' => $ipv4,
+            'port' => $port,
+            'service_name' => $serviceName,
+        ]);
+    }
+
+    private function ipv4ToInt(string $ipv4): int
+    {
+        if ($ipv4 == 'localhost') {
+            $ipv4 = '127.0.0.1';
+        } elseif ($ipv4 == '::1') {
+            $ipv4 = '127.0.0.1';
+        }
+
+        $long = ip2long($ipv4);
+        if (PHP_INT_SIZE === 8) {
+            return $long >> 31 ? $long - (1 << 32) : $long;
+        }
+        return $long;
+    }
+
+    // Used for Zipkin binary annotations like CA/SA (client/server address).
+    // They are modeled as Boolean type with '0x01' as the value.
+    private function makePeerAddressTag(string $key, Endpoint $host): BinaryAnnotation
+    {
+        return new BinaryAnnotation([
+            "key" => $key,
+            "value" => '0x01',
+            "annotation_type" => AnnotationType::BOOL,
+            "host" => $host,
+        ]);
+    }
+
+    /**
+     * @param JaegerThriftSpan[] $spans
+     */
+    private function send(array $spans)
+    {
+        /** @var Tag[] $tags */
+        $tags = [];
+
+        // append tracer global tags to process tags
+        foreach ($this->tracer->getTags() as $k => $v) {
+            if ($k === \Jaeger\JAEGER_HOSTNAME_TAG_KEY) {
+                $k = "hostname";
+            }
+            $tags[] = new Tag([
+                "key" => $k,
+                "vType" => TagType::STRING,
+                "vStr" => $v
+            ]);
+        }
+
+        $tags[] = new Tag([
+            "key" => "format",
+            "vType" => TagType::STRING,
+            "vStr" => "jaeger.thrift"
+        ]);
+
+        $tags[] = new Tag([
+            "key" => "ip",
+            "vType" => TagType::STRING,
+            "vStr" => $this->tracer->getIpAddress()
+        ]);
+
+        $batch = new Batch([
+            "spans" => $spans,
+            "process" => new Process([
+                "serviceName" => $this->tracer->getServiceName(),
+                "tags" => $tags
+            ])
+        ]);
+
+        $this->agentClient->emitBatch($batch);
+    }
+
+    /**
+     * @param JaegerSpan $span
+     */
+    public function append(JaegerSpan $span)
+    {
+        $this->spans[] = $span;
+    }
+
+    public function close()
```

Comment:
This seems to be unused.

Reply:
I made it similar to UdpSender: http://github.com/pekhota/jaeger-client-php/blob/edcb9bbbc813ff0f329a1c1877600a1f3aac5194/src/Jaeger/Sender/UdpSender.php#L103-L103
It was used in Remote reporter, so I made an interface with all the methods http://github.com/pekhota/jaeger-client-php/blob/edcb9bbbc813ff0f329a1c1877600a1f3aac5194/src/Jaeger/Reporter/RemoteReporter.php#L46-L46 and implemented it as empty method.